torchaudio/datasets/utils.py

```diff
@@ -76,41 +65,85 @@ def makedir_exist_ok(dirpath):
             raise
 
 
-def download_url(url, root, filename=None, md5=None):
-    """Download a file from a url and place it in root.
+def download_url(url, download_folder, hash_value=None, hash_type="sha256"):
+    """Execute the correct download operation.
+    Depending on the size of the file online and offline, resume the
```

Comment:
I'd make resuming an option, similar to how something like rsync allows you to decide whether you want to just simply overwrite or attempt to detect differences.

Reply:
Sure